### PR TITLE
print traces when transforming an asset

### PR DESCRIPTION
**Pull request by andrewkolos**
From https://github.com/flutter/flutter/issues/143348#issuecomment-2016047148:

> before we ship, we should add a printTrace to the tool about each asset transformer we're invoking and the path/arguments we called it with

I think this is a good idea since asset transformers can be arbitrary Dart programs—meaning that a lot can go wrong when running them. For example, they can hang indefinitely or perform some sort of I/O that later results in a tool crash. Knowing that asset transformation was involved when debugging a crash (or a slow/stuck `flutter build`) could be useful, so I think adding a `printTrace` or two is a good idea (or at least not a bad one).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by andrewkolos** [comment]
Example of what this looks like (excerpt from running `flutter run -vv -d macos` using https://github.com/andrewkolos/asset_transformers_test):

```
[        ] [   +3 ms] /Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/dartaotruntime --disable-dart-dev /Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server_aot.dart.snapshot --sdk-root /Users/andrewkolos/Documents/GitHub/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --target=flutter --no-print-incremental-dependencies -DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/6974dbac35a175a337ec2a56045b00dc55d3e261/ -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --packages /Users/andrewkolos/Desktop/asset_transformers_test/.dart_tool/package_config.json --output-dill /Users/andrewkolos/Desktop/asset_transformers_test/.dart_tool/flutter_build/b69fff6f08b8bfabd439391362a2e8e7/app.dill --depfile /Users/andrewkolos/Desktop/asset_transformers_test/.dart_tool/flutter_build/b69fff6f08b8bfabd439391362a2e8e7/kernel_snapshot.d --incremental --initialize-from-dill /Users/andrewkolos/Desktop/asset_transformers_test/.dart_tool/flutter_build/b69fff6f08b8bfabd439391362a2e8e7/app.dill --native-assets /Users/andrewkolos/Desktop/asset_transformers_test/.dart_tool/flutter_build/b69fff6f08b8bfabd439391362a2e8e7/native_assets.yaml --verbosity=error package:asset_transformers_test/main.dart
[        ] [+3443 ms] kernel_snapshot: Complete
[        ] [ +471 ms] debug_macos_bundle_flutter_assets: Starting due to {}
[        ] [  +94 ms] shaderc command: [/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/artifacts/engine/darwin-x64/impellerc, --sksl, --runtime-stage-metal, --iplr, --sl=/Users/andrewkolos/Desktop/asset_transformers_test/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag, --spirv=/Users/andrewkolos/Desktop/asset_transformers_test/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag.spirv, --input=/Users/andrewkolos/Documents/GitHub/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag, --input-type=frag, --include=/Users/andrewkolos/Documents/GitHub/flutter/packages/flutter/lib/src/material/shaders, --include=/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/artifacts/engine/darwin-x64/shader_lib]
[        ] [  +18 ms] Transforming asset file at path '/Users/andrewkolos/Desktop/asset_transformers_test/assets/svg.svg'...
[        ] [        ] Transforming asset using command '/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/dart run vector_graphics_compiler --input=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.SdyGAv/svg.svg-transformOutput0.svg --output=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.SdyGAv/svg.svg-transformOutput1.svg'
[        ] [        ] Transforming asset file at path '/Users/andrewkolos/Desktop/asset_transformers_test/assets/sparrow.jpeg'...
[        ] [        ] Transforming asset using command '/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/dart run blur_transformer --input=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.SdyGAv/sparrow.jpeg-transformOutput0.jpeg --output=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.SdyGAv/sparrow.jpeg-transformOutput1.jpeg'
[        ] [        ] Transforming asset file at path '/Users/andrewkolos/Desktop/asset_transformers_test/dependency_with_assets/asset-from-dependency-pubspec.png'...
[        ] [        ] Transforming asset using command '/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/dart run blur_transformer --input=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.SdyGAv/asset-from-dependency-pubspec.png-transformOutput0.png --output=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.SdyGAv/asset-from-dependency-pubspec.png-transformOutput1.png'
[        ] [+15683 ms] Finished transforming asset at path '/Users/andrewkolos/Desktop/asset_transformers_test/assets/svg.svg' (15685ms)
[+1341 ms] [  +16 ms] Finished transforming asset at path '/Users/andrewkolos/Desktop/asset_transformers_test/dependency_with_assets/asset-from-dependency-pubspec.png' (15700ms)
[        ] [ +684 ms] Finished transforming asset at path '/Users/andrewkolos/Desktop/asset_transformers_test/assets/sparrow.jpeg' (16385ms)
[        ] [  +44 ms] debug_macos_bundle_flutter_assets: Complete
[        ] [ +589 ms] Persisting file store
[        ] [   +4 ms] Done persisting file store
[        ] [   +4 ms] build succeeded.
[        ] [   +9 ms] "flutter assemble" took 22,350ms.
```

File: packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart [hunk 1]
```diff
@@ -88,6 +91,7 @@ final class AssetTransformer {
     } finally {
       ErrorHandlingFileSystem.deleteIfExists(tempInputFile);
       ErrorHandlingFileSystem.deleteIfExists(tempOutputFile);
+      logger.printTrace("Finished transforming asset at path '${asset.path}' (${stopwatch.elapsedMilliseconds}ms)");
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart:94]: Can we move this line to right before the `finally`?

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
@andrewkolos what's the status on this PR?

**Pull request merged**